File: Problems/No.1700_Number_of_Students_Unable_to_Eat_Lunch/number_of_students_unable_to_eat_lunch.py
class Solution(object):
    def countStudents(self, students, sandwiches):
        """
        :type students: List[int]
        :type sandwiches: List[int]
        :rtype: int
        """

        # start computing
        while students:
            for sandwich in sandwiches:
                cnt = 0
                while (students[0] != sandwich):
                    students.append(students.pop(0))
                    cnt += 1
                    if (cnt == len(students)):
                        return len(students)
                students.pop(0)
        
        return 0

File: Problems/No.1700_Number_of_Students_Unable_to_Eat_Lunch/number_of_students_unable_to_eat_lunch.cpp
class Solution {
public:
    int countStudents(vector<int>& students, vector<int>& sandwiches) {
        
        // parameter
        std::queue<int> sQueue;

        // start computing
        for (auto &s: students) sQueue.push(s);
        
        int cnt;
        while (!sQueue.empty())
            for (auto &sandwich: sandwiches) {
                
                cnt = 0;
                while (sQueue.front() != sandwich) {

                    sQueue.push(sQueue.front());
                    sQueue.pop();
                    cnt++;
                    if (cnt == sQueue.size()) return sQueue.size();

                }
                sQueue.pop();

            }
        
        return 0;

    }
};
